Templating/Helper/DateTimeHelper.php

```diff
@@ -155,17 +159,21 @@ public function process(\IntlDateFormatter $formatter, \Datetime $date)
     /**
      * Gets a date time instance by a given data and timezone.
      *
-     * @param \Datetime|string|int $data     Value representing date
-     * @param null|string          $timezone Timezone of the date
+     * @param \DateTime|\DateTimeInterface|string|int $data     Value representing date
+     * @param null|string                             $timezone Timezone of the date
      *
-     * @return \Datetime
+     * @return \DateTime
      */
     public function getDatetime($data, $timezone = null)
     {
         if ($data instanceof \DateTime) {
             return $data;
         }
 
+        if ($data instanceof \DateTimeImmutable) {
+            return \DateTime::createFromFormat(\DateTime::ATOM, $data->format(\DateTime::ATOM));
```

Comment:
Unfortunately, `IntlDateFormatter` does not seem to support `DateTimeImmutable` (yet?). Therefore, we have to convert it to `DateTime` anyway. However, I'm not sure where do you want to place the conversion. If we don't do it in `getDatetime()` we have to do it in `process()`.